src/wallet/components/WalletProvider.tsx

```diff
@@ -15,11 +22,29 @@ type WalletProviderReact = {
 export function WalletProvider({ children }: WalletProviderReact) {
   const { chain } = useOnchainKit();
   const [isOpen, setIsOpen] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
+  const { address } = useAccount();
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    setIsClosing(true);
+    setTimeout(() => {
+      setIsOpen(false);
+      setIsClosing(false);
+    }, 300);
+  }, []);
+
   const value = useValue({
-    isOpen,
+    address,
     chain,
+    isOpen,
     setIsOpen,
+    isClosing,
+    setIsClosing,
+    handleClose,
+    containerRef,
   });
```

Comment:
Could you do this on `WalletDropdown` instead? It'd avoid having any mismatches in case we ever update the timing (or allow users to change it themselves)

<div
  className={cn(
    pressable.default,
    color.foreground,
    'absolute right-0 z-10 mt-1 flex w-max min-w-[300px] cursor-default flex-col overflow-hidden rounded-xl',
    isClosing ? 'animate-walletContainerOut' : 'animate-walletContainerIn',
    className,
  )}
  onAnimationEnd={() => {
    if (isClosing) {
      setIsOpen(false);
      setIsClosing(false);
    }
  }}
  data-testid="ockWalletDropdown"
>

Reply:
would an easier fix here be to add a parameter to handleClose called `animationDuration`?

Reply:
Hmm I think that'd still be suboptimal because it still requires you to keep the animation duration of the class and the prop in sync manually. If we use transitionEnd instead, then it'll offload that logic to the browser / React

Reply:
ack, i've refactored to use onAnimationEnd